cmd/sonobuoy/app/splat.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2021 VMware Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/vmware-tanzu/sonobuoy/pkg/errlog"
+
+	"github.com/spf13/cobra"
+)
+
+func NewCmdSplat() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "splat AGGREGATOR_RESULTS_PATH",
+		Short: "Reads all tarballs in the specified directory and splat it to STDOUT (for internal use)",
+		Long:  "Sonobuoy is an introspective kubernetes component that generates reports on cluster conformance, configuration, and more",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := runSplat(args[0]); err != nil {
+				errlog.LogError(err)
+				os.Exit(1)
+			}
+		},
+		Hidden: true,
+		Args:   cobra.ExactArgs(1),
+	}
+	return cmd
+}
+
+func runSplat(dirPath string) error {
+	sonobuoyResults, err := filepath.Glob(filepath.Join(dirPath, "*.tar.gz"))
+	if err != nil {
+		return err
+	}
+
+	// archive all the `files` found above into a single (in-memory) tarball
+	// meaning untar-ing `buffer` below would yield `files`, which are also tarballs
+	buf := bytes.NewBuffer([]byte{})
+
+	if err = loadResults(buf, sonobuoyResults); err != nil {
+		return err
+	}
+
+	fmt.Print(buf.String())
+	return nil
+}
+
+func loadResults(buf *bytes.Buffer, filenames []string) error {
+	gzipWriter := gzip.NewWriter(buf)
+	defer gzipWriter.Close()
+
+	tarWriter := tar.NewWriter(gzipWriter)
+	defer tarWriter.Close()
+
+	for _, filename := range filenames {
+		if err := addFileToTarball(tarWriter, filename); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func addFileToTarball(tarWriter *tar.Writer, filename string) error {
```

Comment:
Ah!, Your tweet makes sense now. Yes I wrote a tarrer and blogged about it cause it's one of those you should do once. After that it's copy and paste.